appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go

```diff
@@ -556,7 +601,7 @@ func (r *EnvironmentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Watches(
 			&source.Kind{Type: &corev1.Secret{}},
 			handler.EnqueueRequestsFromMapFunc(r.findObjectsForSecret),
-			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}), // secret doesn't have a generation, so we use RV
```

Comment:
I wasted several hours trying to figure out why `GenerationChangedPredicate` wasn't work here :grin:.